yarn-project/prover-client/src/orchestrator/tx-proving-state.ts

```diff
@@ -37,11 +40,24 @@ export class TxProvingState {
     let previousKernelProof: Proof | undefined = processedTx.proof;
     let previousProofType = PublicKernelType.NON_PUBLIC;
     for (const kernelRequest of processedTx.publicKernelRequests) {
+      const provingRequest: ProvingRequest =
+        kernelRequest.type === PublicKernelType.TAIL
+          ? {
+              type: ProvingRequestType.PUBLIC_KERNEL_TAIL,
+              kernelType: kernelRequest.type,
+              inputs: kernelRequest.inputs,
+            }
+          : {
+              type: ProvingRequestType.PUBLIC_KERNEL_NON_TAIL,
+              kernelType: kernelRequest.type,
+              inputs: kernelRequest.inputs,
+            };
```

Comment:
I'm not a fan of this. I'll look to refactor these `kernelRequests` into `provingRequests` in the next PR.